typegraph/core/src/runtimes/prisma/type_generation/with_filters.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::{
+    errors::Result,
+    global_store::with_store,
+    t::{self, ConcreteTypeBuilder, TypeBuilder},
+    types::{Type, TypeFun, TypeId},
+};
+
+use super::{TypeGen, TypeGenContext};
+
+pub struct WithFilters {
+    type_id: TypeId,
+    skip_rel: bool,
+}
+
+impl WithFilters {
+    pub fn new(type_id: TypeId, skip_rel: bool) -> Self {
+        Self { type_id, skip_rel }
+    }
+}
+
+impl TypeGen for WithFilters {
+    fn generate(&self, context: &mut TypeGenContext) -> Result<TypeId> {
+        enum PropType {
+            Boolean,
+            Number(NumberType),
+            String,
+            Object,
+        }
+        struct Prop {
+            key: String,
+            type_id: TypeId,
+            prop_type: PropType,
+        }
+
+        let mut props = vec![];
+
+        with_store(|s| -> Result<_> {
+            for (k, id) in self.type_id.as_struct(s).unwrap().data.props.iter() {
+                let mut id: TypeId = (*id).into();
+                let mut ty = s.get_type(id)?;
```

Comment:
Curious, does `ty` refer to `ty` in `[Ty]pe` ?

Reply:
Yes :smile: